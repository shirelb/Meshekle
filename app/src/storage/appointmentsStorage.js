import axios from "axios";
import {APP_SOCKET, SERVER_URL} from "../shared/constants";


var getUserAppointmentRequests = function (userId, userHeaders) {
    return axios.get(`${SERVER_URL}/api/appointmentRequests/user/userId/${userId}`,
        {
            headers: userHeaders,
            params: {
                status: 'requested'
            },
        })
        .then(response => {
            return response;
        })
        .catch(error => {
            console.log('get user appointment requests error ', error)
        });
};

var getUserAppointments = function (userId, userHeaders) {
    return axios.get(`${SERVER_URL}/api/appointments/user/userId/${userId}`,
        {
            headers: userHeaders,
            params: {
                status: 'set'
            },
        })
        .then(response => {
            return response;
        })
        .catch(error => {
            console.log('get user appointments error ', error)
        });
};

var getUserAppointmentById = function (userId, userHeaders, eventId) {
    return axios.get(`${SERVER_URL}/api/appointments/user/userId/${userId}`,
        {
            headers: userHeaders,
            params: {status: 'set', appointmentId: eventId}
        })
        .then(response => {
            return response;
        })
        .catch(error => {
            console.log('get user appointments by id error ', error)
        });
};

var postUserAppointmentRequest = function (userId, serviceProvider, appointmentRequest, userHeaders) {
    return axios.post(`${SERVER_URL}/api/appointmentRequests/user/request`,
        {
            userId: userId,
            serviceProviderId: serviceProvider.serviceProviderId,
            role: serviceProvider.role,
            availableTime: appointmentRequest.availableTime,
            notes: appointmentRequest.notes ? appointmentRequest.notes : '',
            subject: JSON.stringify(appointmentRequest.subject)
        },
        {
            headers: userHeaders,
        })
        .then(response => {
            APP_SOCKET.emit('userPostAppointmentRequests', {
                serviceProviderId: serviceProvider.serviceProviderId
            });

            return response;
        })
        .catch(error => {
            console.log('post user appointment request error ', error)
        });
};

var cancelAppointmentRequestById = (appointmentRequest, headers) => {
    return axios.put(`${SERVER_URL}/api/appointmentRequests/user/reject`,
        {
            userId: appointmentRequest.AppointmentDetail.clientId,
            appointmentRequestId: appointmentRequest.requestId,
        },
        {
            headers: headers,
            params: {
                status: 'rejected'
            },
        }
    )
        .then((response) => {
            APP_SOCKET.emit('userCancelAppointmentRequests', {
                serviceProviderId: appointmentRequest.AppointmentDetail.serviceProviderId,
            });

            return response
        })
        .catch((error) => {
            console.log('reject appointment request error ', error);
        });
};

var cancelAppointmentById = (appointment, headers) => {
    return axios.put(`${SERVER_URL}/api/appointments/user/cancel/userId/${appointment.AppointmentDetail.clientId}/appointmentId/${appointment.appointmentId}`,
        {},
        {
            headers: headers
        }
    )
        .then((response) => {
            APP_SOCKET.emit('userCancelAppointment', {
                serviceProviderId: appointment.AppointmentDetail.serviceProviderId,
            });

            return response
        })
        .catch((error) => {
            console.log('cancel appointment error ', error);
        });
};


export default {
    getUserAppointments,
    getUserAppointmentById,
    postUserAppointmentRequest,
    getUserAppointmentRequests,
    cancelAppointmentRequestById,
    cancelAppointmentById,
};
