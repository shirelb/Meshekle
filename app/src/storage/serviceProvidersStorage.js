import axios from "axios";
import {SERVER_URL} from "../shared/constants";

var getServiceProviders = function (userHeaders) {
    return axios.get(`${SERVER_URL}/api/serviceProviders`,
        {headers: userHeaders}
    )
        .then(response => {
            return response;
        })
        .catch(error => {
            console.log('load appointments error ', error)
        });
};


export default {getServiceProviders};
