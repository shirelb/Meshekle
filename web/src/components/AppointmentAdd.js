import React from 'react';
import axios, {post} from 'axios';
import {Helmet} from 'react-helmet';
import AppointmentForm from "./AppointmentForm";
import {Grid, Header, Modal} from "semantic-ui-react";
import store from "store";
import helpers from "../shared/helpers";
import mappers from "../shared/mappers";
import {SERVER_URL} from "../shared/constants";

class AppointmentAdd extends React.Component {
    constructor(props) {
        super(props);

        this.state = {slotInfo: this.props.location.state.slotInfo};

        this.handleSubmit = this.handleSubmit.bind(this);
        this.handleCancel = this.handleCancel.bind(this);
    }

    componentDidMount() {
        this.serviceProviderHeaders = {
            'Authorization': 'Bearer ' + store.get('serviceProviderToken')
        };
    }

    handleSubmit(appointment) {
        let appointmentToSend = {
            userId: appointment.clientId,
            serviceProviderId: store.get('serviceProviderId'),
            // role: 'Driver',
            date: appointment.date,
            startHour: appointment.startTime,
            endHour: appointment.endTime,
            notes: appointment.remarks,
            subject: appointment.subject
        };

        let serviceProviderRoles;
        helpers.getRolesOfServiceProvider(store.get('serviceProviderId'))
            .then(roles => {
                serviceProviderRoles = roles.map(role => mappers.rolesMapper(role));

                appointmentToSend.role = serviceProviderRoles[0];

                axios.post(`${SERVER_URL}/api/users/appointments/set`,
                    {
                        data: appointmentToSend,
                        headers: this.serviceProviderHeaders
                    }
                )
                    .then((response) => {
                        console.log('add appointment ', response);
                    })
                    .catch((error) => {
                        console.log('add appointment ', error);
                    });
            })
            .catch(error => console.log('error ', error));


        /*post('/api/appointments', appointment)
            .then(() => {
                console.log('added:', appointment);
            });*/
        console.log('appoitment ', appointment)
    }

    handleCancel(e) {
        e.preventDefault();

        this.props.history.goBack()
    }

    render() {
        return (
            <Modal size='small' open dimmer="blurring" closeIcon onClose={() => this.props.history.goBack()}>
                {/*<Page title="Add Appointment" columns={3}>*/}
                <Helmet>
                    <title>Meshekle | Add Appointment</title>
                </Helmet>

                <Grid padded>
                    <Grid.Column>

                        <Header as="h1" floated="right">תור חדש</Header>

                        <AppointmentForm
                            handleSubmit={this.handleSubmit}
                            handleCancel={this.handleCancel}
                            slotInfo={this.state.slotInfo}
                        />
                    </Grid.Column>
                </Grid>
                {/*</Page>*/}
            </Modal>
        );
    }
}

export default AppointmentAdd;