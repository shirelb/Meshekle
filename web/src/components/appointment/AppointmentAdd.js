import React from 'react';
import axios, {post} from 'axios';
import {Helmet} from 'react-helmet';
import AppointmentForm from "./AppointmentForm";
import {Grid, Header, Modal} from "semantic-ui-react";
import store from "store";
import moment from 'moment';
import helpers from "../../shared/helpers";
import mappers from "../../shared/mappers";
import {SERVER_URL} from "../../shared/constants";

class AppointmentAdd extends React.Component {
    constructor(props) {
        super(props);

        this.state = {slotInfo: this.props.location.state.slotInfo};

        this.handleSubmit = this.handleSubmit.bind(this);
        this.handleCancel = this.handleCancel.bind(this);
    }

    componentDidMount() {
        this.serviceProviderHeaders = {
            'Authorization': 'Bearer ' + store.get('serviceProviderToken')
        };
    }

    handleSubmit(appointment) {
        helpers.getRolesOfServiceProvider(store.get('serviceProviderId'))
            .then(roles => {
                let serviceProviderRoles = roles.map(role => mappers.rolesMapper(role));

                axios.post(`${SERVER_URL}/api/serviceProviders/appointments/set`,
                    {
                        userId: appointment.clientId,
                        serviceProviderId: store.get('serviceProviderId'),
                        role: serviceProviderRoles[0],
                        date: moment.isMoment(appointment.date) ? appointment.date.format('YYYY-MM-DD') : appointment.date,
                        startHour: moment.isMoment(appointment.startTime) ? appointment.startTime.format("HH:mm") : appointment.startTime,
                        endHour: moment.isMoment(appointment.endTime) ? appointment.endTime.format("HH:mm") : appointment.endTime,
                        notes: appointment.remarks ? appointment.remarks : '',
                        subject: JSON.stringify(appointment.subject)
                    },
                    {
                        headers: this.serviceProviderHeaders
                    }
                )
                    .then((response) => {
                        console.log('add appointment ', response);
                        this.props.history.goBack()
                    })
                    .catch((error) => {
                        console.log('add appointment ', error);
                    });
            })
            .catch(error => console.log('error ', error));
    }

    handleCancel(e) {
        e.preventDefault();

        this.props.history.goBack()
    }

    render() {
        return (
            <Modal size='small' open dimmer="blurring" closeIcon onClose={() => this.props.history.goBack()}>
                {/*<Page title="Add Appointment" columns={3}>*/}
                <Helmet>
                    <title>Meshekle | Add Appointment</title>
                </Helmet>

                <Grid padded>
                    <Grid.Column>

                        <Header as="h1" floated="right">תור חדש</Header>

                        <AppointmentForm
                            submitText="קבע"
                            handleSubmit={this.handleSubmit}
                            handleCancel={this.handleCancel}
                            slotInfo={this.state.slotInfo}
                            // appointment={}
                        />
                    </Grid.Column>
                </Grid>
                {/*</Page>*/}
            </Modal>
        );
    }
}

export default AppointmentAdd;