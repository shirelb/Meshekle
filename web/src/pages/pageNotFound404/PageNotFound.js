import React from 'react';
import {Grid, Header, Image} from 'semantic-ui-react';

const PageNotFound = () => (
    <Grid stretched textAlign={"center"}>
        <Grid.Column stretched textAlign={"center"}>
            <Image centered src={require('../../images/logo1.png')} size={'large'}/>
            <Header size={"huge"} textAlign={"center"} icon style={{marginTop: "-10px"}}>
                משקל'ה
                <Header.Subheader>יש משק ויש משקל'ה</Header.Subheader>
            </Header>

            <div style={{marginTop: 20, textAlign: "center"}}>
                <Header as={'h1'}> 404 </Header>
                <Header as={'h2'}> Page not found </Header>
            </div>
        </Grid.Column>
    </Grid>
);

export default PageNotFound;