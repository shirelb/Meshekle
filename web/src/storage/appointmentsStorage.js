import axios from "axios";
import {SERVER_URL, WEB_SOCKET} from "../shared/constants";
import moment from "moment";


var getAppointmentByAppointmentID = (serviceProviderId, appointmentId, headers) => {
    return axios.get(`${SERVER_URL}/api/appointments/serviceProvider/serviceProviderId/${serviceProviderId}`,
        {
            headers: headers,
            params: {
                appointmentId: appointmentId,
            }
        }
    )
        .then((response) => {
            let appointment = response.data[0];
            // console.log('getAppointmentByAppointmentID ', appointmentId, ' ', appointment);
            return appointment;
        })
        .catch((error) => {
            console.log('getAppointmentByAppointmentID ', appointmentId, ' ', error);
            return error;
        });
};

var getAppointmentRequestByAppointmentRequestID = (serviceProviderId, appointmentRequestId, headers) => {
    return axios.get(`${SERVER_URL}/api/appointmentRequests/serviceProvider/serviceProviderId/${serviceProviderId}`,
        {
            headers: headers,
            params: {
                appointmentRequestId: appointmentRequestId,
            }
        }
    )
        .then((response) => {
            let appointmentRequest = response.data[0];
            // console.log('getAppointmentByAppointmentID ', appointmentRequestId, ' ', appointmentRequest);
            return appointmentRequest;
        })
        .catch((error) => {
            console.log('getAppointmentByAppointmentID ', appointmentRequestId, ' ', error);
            return error;
        });
};

var getServiceProviderAppointmentRequests = (serviceProviderId, headers) => {
    return axios.get(`${SERVER_URL}/api/appointmentRequests/serviceProvider/serviceProviderId/${serviceProviderId}`,
        {
            headers: headers,
            params: {
                status: 'requested'
            },
        }
    )
        .then((response) => {
            return response;
        })
        .catch((error) => {
            console.log('get service provider appointment requests error ', error);
            return error;
        });
};

var getServiceProviderAppointments = (serviceProviderId, headers) => {
    return axios.get(`${SERVER_URL}/api/appointments/serviceProvider/serviceProviderId/${serviceProviderId}`,
        {
            headers: headers,
            params: {
                status: 'set'
            },
        }
    )
        .then((response) => {
            return response;
        })
        .catch((error) => {
            console.log('get service provider appointment requests error ', error);
            return error;
        });
};

var setAppointment = (appointment, serviceProviderId, roles, headers) => {
    return axios.post(`${SERVER_URL}/api/appointments/serviceProvider/set`,
        {
            userId: appointment.clientId,
            serviceProviderId: serviceProviderId,
            role: appointment.role,
            date: moment.isMoment(appointment.date) ? appointment.date.format('YYYY-MM-DD') : appointment.date,
            startHour: moment.isMoment(appointment.startTime) ? appointment.startTime.format("HH:mm") : appointment.startTime,
            endHour: moment.isMoment(appointment.endTime) ? appointment.endTime.format("HH:mm") : appointment.endTime,
            notes: appointment.remarks ? appointment.remarks : '',
            subject: JSON.stringify(appointment.subject)
        },
        {
            headers: headers
        }
    )
        .then((response) => {
            WEB_SOCKET.emit('serviceProviderPostAppointment', {
                userId: appointment.clientId,
            });

            return response;
        })
        .catch((error) => {
            console.log('add appointment ', error);
            return error;
        });
};

var updateAppointment = (event, headers) => {
    return axios.put(`${SERVER_URL}/api/appointments/serviceProvider/update/appointmentId/${event.appointmentId}`,
        {
            startDateAndTime: moment(event.date + ' ' + event.startTime).toDate(),
            endDateAndTime: moment(event.date + ' ' + event.endTime).toDate(),
            remarks: event.remarks,

            subject: JSON.stringify(event.subject),
            clientId: event.clientId,
            role: event.role,
        },
        {
            headers: headers,
        }
    )
        .then((response) => {
            WEB_SOCKET.emit('serviceProviderUpdateAppointment', {
                userId: event.clientId
            });

            return response;
        })
        .catch((error) => {
            console.log('update appointment error ', error);
            return error;
        });
};

var cancelAppointmentById = (appointment, headers) => {
    return axios.put(`${SERVER_URL}/api/appointments/serviceProvider/cancel/appointmentId/${appointment.appointmentId}`,
        {},
        {
            headers: headers
        }
    )
        .then((response) => {
            WEB_SOCKET.emit('serviceProviderCancelAppointment', {
                userId: appointment.AppointmentDetail.clientId,
            });

            return response
        })
        .catch((error) => {
            console.log('cancel appointment error ', error);
            return error;
        });
};

var rejectAppointmentRequestById = (appointmentRequest, headers) => {
    return axios.put(`${SERVER_URL}/api/appointmentRequests/serviceProvider/update/status/appointmentRequestId/${appointmentRequest.requestId}`,
        {},
        {
            headers: headers,
            params: {
                status: 'rejected'
            },
        }
    )
        .then((response) => {
            WEB_SOCKET.emit('serviceProviderRejectAppointmentRequest', {
                userId: appointmentRequest.AppointmentDetail.clientId,
            });

            return response
        })
        .catch((error) => {
            console.log('reject appointment request error ', error);
            return error;
        });
};

var approveAppointmentRequestById = (appointmentRequest, headers) => {
    return axios.put(`${SERVER_URL}/api/appointmentRequests/serviceProvider/update/status/appointmentRequestId/${appointmentRequest.requestId}`,
        {},
        {
            headers: headers,
            params: {
                status: 'approved'
            },
        }
    )
        .then((response) => {
            WEB_SOCKET.emit('serviceProviderApproveAppointmentRequest', {
                userId: appointmentRequest.AppointmentDetail.clientId,
            });

            return response
        })
        .catch((error) => {
            console.log('approve appointment request error ', error);
            return error;
        });
};


export default {
    getAppointmentByAppointmentID,
    getAppointmentRequestByAppointmentRequestID,
    setAppointment,
    cancelAppointmentById,
    rejectAppointmentRequestById,
    approveAppointmentRequestById,
    getServiceProviderAppointmentRequests,
    getServiceProviderAppointments,
    updateAppointment,
};
