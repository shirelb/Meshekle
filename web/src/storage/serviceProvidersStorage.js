import axios from "axios";
import {SERVER_URL} from "../shared/constants";
import store from "store";

const serviceProviderHeaders = {
    'Authorization': 'Bearer ' + store.get('serviceProviderToken')
};

var getServiceProviders = function () {
    return axios.get(`${SERVER_URL}/api/serviceProviders`,
        {headers: serviceProviderHeaders}
    )
        .then(response => {
            return response;
        })
        .catch(error => {
            console.log('get serviceProviders error ', error)
        });
};

var getRolesOfServiceProvider = (serviceProviderId) => {
    return axios.get(`${SERVER_URL}/api/serviceProviders/roles/serviceProviderId/${serviceProviderId}`,
        {headers: serviceProviderHeaders}
    )
        .then((response) => {
            let roles = response.data;
            console.log('roles ', roles);
            return roles;
        })
        .catch((error) => {
            console.log('get serviceProvider roles error ', error)
        });
};

var serviceProviderLogin = (userId, password) => {
    return axios.post(`${SERVER_URL}/api/serviceProviders/login/authenticate`,
        {
            "userId": userId,
            "password": password
        },
    );
};

var serviceProviderValidToken = function (token) {
    return axios.post(`${SERVER_URL}/api/serviceProviders/validToken`,
        {
            "token": token
        },
    );
}

var getServiceProviderPermissionsById = (serviceProviderId) => {
    return axios.get(`${SERVER_URL}/api/serviceProviders/serviceProviderId/${serviceProviderId}/permissions`,
        {headers: serviceProviderHeaders}
    )
        .then((response) => {
            let permissions = response.data;
            console.log('permissions ', permissions);
            return permissions;
        })
        .catch((error) => {
            console.log('error ', error);
        });
};


export default {
    getServiceProviders,
    getRolesOfServiceProvider,
    serviceProviderLogin,
    serviceProviderValidToken,
    getServiceProviderPermissionsById,

};
