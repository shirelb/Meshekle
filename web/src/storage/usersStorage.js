import axios from "axios";
import {SERVER_URL, WEB_SOCKET} from "../shared/constants";
import store from "store";

const getServiceProviderHeaders = () => {
    return {
        'Authorization': 'Bearer ' + store.get('serviceProviderToken')
    };
};

var getUserByUserID = (userId, headers = null) => {
    return axios.get(`${SERVER_URL}/api/users/userId/${userId}`,
        {headers: headers ? headers : getServiceProviderHeaders()}
    )
        .then((response) => {
            return response.data[0];
        })
        .catch((error) => {
            console.log('getUserByUserID ', userId, ' error ', error);
            return error;
        });
};

var getUsers = (headers = null) => {
    return axios.get(`${SERVER_URL}/api/users`,
        {headers: headers ? headers : getServiceProviderHeaders()}
    )
        .then((response) => {
            return response.data.filter(user => user.userId !== 1 && user.userId !== "1");
        })
        .catch((error) => {
            console.log('getUsers error ', error);
            return error;
        });
};

var deleteUserByUserID = (userId, headers = null) => {
    return axios.delete(`${SERVER_URL}/api/serviceProviders/users/userId/${userId}/delete`,
        {headers: headers ? headers : getServiceProviderHeaders()}
    )
        .then((response) => {
            WEB_SOCKET.emit('userShallowDeleted');

            return response;
        })
        .catch((error) => {
            console.log('deleteUserByUserID ', userId, ' error ', error);
            return error;
        });
};

var createUser = (newUser, serviceProviderHeaders) => {
    serviceProviderHeaders['Access-Control-Allow-Origin'] = '*';
    return axios.post(`${SERVER_URL}/api/serviceProviders/users/add`,
        {
            userId: newUser.userId,
            fullname: newUser.fullname,
            email: newUser.email,
            mailbox: newUser.mailbox,
            cellphone: newUser.cellphone,
            phone: newUser.phone,
            bornDate: newUser.bornDate,
            image: newUser.image ? newUser.image : null,
        },
        {headers: serviceProviderHeaders}
    )
        .then((response) => {
            WEB_SOCKET.emit('userCreated');

            return response;
        })
        .catch((error) => {
            console.log('createUser error ', error);
            return error;
        });
};

var updateUserById = function (updatedUser, serviceProviderHeaders) {
    return axios.put(`${SERVER_URL}/api/users/update/userId/${updatedUser.userId}`,
        {
            fullname: updatedUser.fullname ? updatedUser.fullname : null,
            password: updatedUser.password ? updatedUser.password : null,
            email: updatedUser.email ? updatedUser.email : null,
            mailbox: updatedUser.mailbox ? updatedUser.mailbox : null,
            cellphone: updatedUser.cellphone ? updatedUser.cellphone : null,
            phone: updatedUser.phone ? updatedUser.phone : null,
            bornDate: updatedUser.bornDate ? updatedUser.bornDate : null,
            active: typeof updatedUser.active === 'boolean' ? updatedUser.active : null,
            image: updatedUser.image ? updatedUser.image : null,
        },
        {headers: serviceProviderHeaders}
    )
        .then(response => {
            WEB_SOCKET.emit('userUpdated');

            return response;
        })
        .catch(error => {
            console.log('updateUserById error ', error);
            return error;
        });
};


export default {getUserByUserID, getUsers, deleteUserByUserID, createUser, updateUserById};
