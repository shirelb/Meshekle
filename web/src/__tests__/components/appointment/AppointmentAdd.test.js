import React from 'react';
import {shallow} from "enzyme/build";
import {flushPromises,setupComponent} from "../../testHelpers";
import PageNotFound from "../../../pages/pageNotFound404/PageNotFound";
import store from 'store';
import moment from 'moment';
import {createMemoryHistory} from 'history';
import AppointmentAdd from "../../../components/appointment/AppointmentAdd";
import serviceProvidersStorage from "../../../storage/serviceProvidersStorage";
import appointmentsStorage from "../../../storage/appointmentsStorage";
import usersStorage from "../../../storage/usersStorage";
import users from "../../jsons/users";
import serviceProviders from "../../jsons/serviceProviders";
import appointmentsOf549963652 from "../../jsons/appointmentsServiceProvider549963652";

const appointmentRequestsOf549963652 = require("../../jsons/appointmentRequestsServiceProvider549963652")

jest.mock("store");
jest.mock("../../../storage/usersStorage");
jest.mock("../../../storage/serviceProvidersStorage");
jest.mock("../../../storage/appointmentsStorage");


describe("AppointmentAdd should", () => {
    let wrapper = null;
    let componentInstance = null;
    let appointmentRequestTest = appointmentRequestsOf549963652[1];
    appointmentRequestTest.optionalTimes = JSON.parse(appointmentRequestTest.optionalTimes);
    const history = createMemoryHistory({
        initialEntries: ['/', '/home', '/appointments', '/appointments/set'],
        initialIndex: 3,
        keyLength: 10,
        getUserConfirmation: null
    });
    const slotInfo = {
        end: moment(),
        start: moment(),
    };
    const appointmentRequestDropped = {
        allDay: false,
        appointmentRequest: appointmentRequestTest,
        end: moment(),
        id: 985,
        start: moment(),
        title: "Catherina Blackmuir",
    };
    const propsWithSlotInfo = {
        history: history,
        location: {
            pathname: '/appointments/set',
            state: {
                slotInfo: slotInfo
            }
        },
        match: {
            isExact: true,
            path: '/appointments/set',
            url: '/appointments/set',
        },
        serviceProviderRoles: ["appointmentsHairDresser"],
        getUsersForAppointmentForm: jest.fn(),
        getServiceProviderRoles: jest.fn(),
        userOptions: [],
    };
    const propsWithRequest = {
        history: history,
        location: {
            pathname: '/appointments/set',
            state: {
                appointmentRequestDropped: appointmentRequestDropped
            }
        },
        match: {
            isExact: true,
            path: '/appointments/set',
            url: '/appointments/set',
        },
        serviceProviderRoles: ["appointmentsHairDresser"],
        getUsersForAppointmentForm: jest.fn(),
        getServiceProviderRoles: jest.fn(),
        userOptions: [],
    };
    const mockStore = {
        serviceProviderId: "549963652",
        userId: "549963652",
    };

    store.get.mockImplementation((key) => mockStore[key]);
    usersStorage.getUsers.mockResolvedValue(users);
    usersStorage.getUserByUserID.mockImplementation((userId) => Promise.resolve(users.filter(user => user.userId === userId)[0]));
    serviceProvidersStorage.getServiceProviderUserDetails.mockImplementation((serviceProviderId) => Promise.resolve({data: users.filter(user => user.userId === serviceProviderId)[0]}));
    serviceProvidersStorage.getServiceProviderById.mockImplementation((serviceProviderId) => Promise.resolve(serviceProviders.filter(provider =>
        provider.serviceProviderId === serviceProviderId)));
    serviceProvidersStorage.getRolesOfServiceProvider.mockResolvedValue(["appointmentsHairDresser"]);
    appointmentsStorage.setAppointment.mockResolvedValue({});
    appointmentsStorage.getServiceProviderAppointments.mockResolvedValue({data: appointmentsOf549963652});
    appointmentsStorage.getServiceProviderAppointmentRequests.mockImplementation(() => Promise.resolve({data: JSON.parse(JSON.stringify(appointmentRequestsOf549963652))}));

    beforeAll(() => {
    });

    afterAll(() => {
    });

    beforeEach(async () => {
        jest.clearAllMocks();
        await flushPromises();
    });

    afterEach(() => {
    });

    test('match snapshot with slotInfo', async () => {
        const props = {
            location: {
                pathname: '/appointments/set',
                state: {
                    slotInfo: slotInfo
                }
            },
            match: {
                isExact: true,
                path: '/appointments/set',
                url: '/appointments/set',
            },
            serviceProviderRoles: ["appointmentsHairDresser"],
            getUsersForAppointmentForm: jest.fn(),
            getServiceProviderRoles: jest.fn(),
            userOptions: [],
        };

        const arrResponse = setupComponent('shallow', AppointmentAdd, null, props, "/appointments/set");
        wrapper = arrResponse[0];
        componentInstance = arrResponse[1];

        expect(componentInstance).toMatchSnapshot();
    });

    test('match snapshot with appointmentRequestDropped', async () => {
        const props = {
            location: {
                pathname: '/appointments/set',
                state: {
                    appointmentRequestDropped: appointmentRequestDropped
                }
            },
            match: {
                isExact: true,
                path: '/appointments/set',
                url: '/appointments/set',
            },
            serviceProviderRoles: ["appointmentsHairDresser"],
            getUsersForAppointmentForm: jest.fn(),
            getServiceProviderRoles: jest.fn(),
            userOptions: [],
        };

        const arrResponse = setupComponent('shallow', AppointmentAdd, null, props, "/appointments/set");
        wrapper = arrResponse[0];
        componentInstance = arrResponse[1];

        expect(componentInstance).toMatchSnapshot();
    });

    test("renders AppointmentAdd for /appointments/set and slotInfo", async () => {
        const arrResponse = await setupComponent('shallow', AppointmentAdd, history, propsWithSlotInfo, "/appointments/set");
        wrapper = arrResponse[0];
        componentInstance = arrResponse[1];

        expect(wrapper.find(AppointmentAdd)).toHaveLength(1);
        expect(wrapper.find(PageNotFound)).toHaveLength(0);
    });

    test("mounted with the right data for slotInfo", async () => {
        const arrResponse = await setupComponent('shallow', AppointmentAdd, history, propsWithSlotInfo, "/appointments/set");
        wrapper = arrResponse[0];
        componentInstance = arrResponse[1];

        expect(componentInstance.state.slotInfo).toBeDefined();
        expect(componentInstance.state.slotInfo.end).toEqual(slotInfo.end);
        expect(componentInstance.state.slotInfo.start).toEqual(slotInfo.start);
    });

    test("renders AppointmentAdd for /appointments/set and appointmentRequestDropped", async () => {
        const arrResponse = await setupComponent('shallow', AppointmentAdd, history, propsWithRequest, "/appointments/set");
        wrapper = arrResponse[0];
        componentInstance = arrResponse[1];

        expect(wrapper.find(AppointmentAdd)).toHaveLength(1);
        expect(wrapper.find(PageNotFound)).toHaveLength(0);
    });

    test("mounted with the right data for appointmentRequestDropped", async () => {
        const arrResponse = await setupComponent('shallow', AppointmentAdd, history, propsWithRequest, "/appointments/set");
        wrapper = arrResponse[0];
        componentInstance = arrResponse[1];

        expect(componentInstance.state.appointmentRequestEvent).toBeDefined();
        expect(componentInstance.state.appointmentRequestEvent.appointmentRequest).toBeDefined();
        expect(componentInstance.state.appointmentRequestEvent.allDay).toEqual(appointmentRequestDropped.allDay);
        expect(componentInstance.state.appointmentRequestEvent.end).toEqual(appointmentRequestDropped.end);
        expect(componentInstance.state.appointmentRequestEvent.start).toEqual(appointmentRequestDropped.start);
        expect(componentInstance.state.appointmentRequestEvent.title).toEqual(appointmentRequestDropped.title);
        expect(componentInstance.state.appointmentRequestEvent.id).toEqual(appointmentRequestDropped.id);
    });

    test("handleCancel", async () => {
        const arrResponse = await setupComponent('mount', AppointmentAdd, history, propsWithRequest, "/appointments/set");
        wrapper = arrResponse[0];
        componentInstance = arrResponse[1];

        await componentInstance.handleCancel({
            preventDefault() {
            }
        });

        expect(componentInstance.props.history.location.pathname).toEqual('/appointments');
    });

    test("render with what the user see", async () => {
        const arrResponse = await setupComponent('shallow', AppointmentAdd, history, propsWithSlotInfo, "/appointments/set");
        wrapper = arrResponse[0];
        componentInstance = arrResponse[1];

        const shallowWrapper = wrapper.find('AppointmentAdd').dive();

        expect(shallowWrapper.find('Modal')).toHaveLength(1);
        expect(shallowWrapper.find('Grid')).toHaveLength(1);
        expect(shallowWrapper.find('Header')).toHaveLength(1);
        expect(shallowWrapper.find('Header').props().children).toEqual('תור חדש');
        expect(shallowWrapper.find('AppointmentForm')).toHaveLength(1);
    });

    test("handleSubmit", async () => {
        const arrResponse = await setupComponent('mount', AppointmentAdd, history, propsWithSlotInfo, "/appointments/set");
        wrapper = arrResponse[0];
        componentInstance = arrResponse[1];

        await componentInstance.handleSubmit(appointmentsOf549963652[1]);
        expect(componentInstance.props.history.location.pathname).toEqual('/home');
    });
});
