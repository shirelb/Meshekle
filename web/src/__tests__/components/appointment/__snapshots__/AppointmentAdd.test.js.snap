// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppointmentAdd should match snapshot with appointmentRequestDropped 1`] = `undefined`;

exports[`AppointmentAdd should match snapshot with slotInfo 1`] = `undefined`;
